03/Product.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+class Product extends React.Component {
+	state = {
+		isDisable: false,
+	};
+
+	handleToggleDisable = () => {
+		const { isDisable } = this.state;
+		this.setState(prevState => ({ isDisable: !isDisable }));
+	};
+	render() {
+		const { product, isCategory, isCart, onAddToCart, onRemoveFromCart } = this.props;
+		return (
+			<div className='product'>
+				<h3>{product.name}</h3>
+				<p>{product.description}</p>
+				<p>Price: ${product.price}</p>
+
+				{isCategory && (
+					<button
+						onClick={() => {
+							onAddToCart(product);
+							this.handleToggleDisable();
+						}}
+						disabled={this.state.isDisable}
+					>
+						+
+					</button>
+				)}
+
+				{isCart && (
+					<button
+						onClick={() => {
```

Comment:
Zwróć uwagę, że tutaj dużo elementów się powtarza.
Może gdybyś przez `props` przekazywał dane o tych samych nazwach to kod by się uprościł.

np. ustalamy, że przycisk może być albo w koszyku, albo nie jest w koszyku (w tedy jest w kategorii) czyli informacja `isCart=true/false` wystarczy. Podobnie z metodą - czy nie możmy wykorzystać tej samej nazwy w props tylko przekazywać inną metodę np. np. `handleClick`:
1) `<Product key={cartItem.id} product={cartItem} isCart={true} handleClick={onRemoveFromCart} />`
2) `<Product key={product.id} product={product} isCart={false} handleClick={onAddToCart} />`

Reply:
Już dopisuję, powyżej miałeś przykład renderowania komponentu, poniże przedstawiam jak to napisać:

```
<button
	onClick={() => {
		this.props.handleClick(product);
		this.handleToggleDisable();
	}}
	disabled={this.state.isDisable}
>
    {isCart ? '+' : '-'}
</button>
```

W tym rozwiązaniu pozbywamy się zduplikowanego kodu - nie musimy robić warunków dla całego `<button>`

Reply:
a czy stan z produktu(isDisable) powinienem wyrzucic do komponentu rodzica tzn app i stamtąd renderowac czy przycisk jest widoczny? bo aktualnie jak usune z koszyka to nie chce wyrenderowac disable z false - czyli nadal mam niewidoczny przycisk '+' ?